storage/filesystem/dotgit/dotgit.go

```diff
@@ -780,33 +788,23 @@ func (d *DotGit) RemoveRef(name plumbing.ReferenceName) error {
 }
 
 func (d *DotGit) addRefsFromPackedRefs(refs *[]*plumbing.Reference, seen map[plumbing.ReferenceName]bool) (err error) {
-	packedRefs, err := d.findPackedRefs()
-	if err != nil {
-		return err
-	}
-
-	for _, ref := range packedRefs {
+	return d.findPackedRefs(func(ref *plumbing.Reference) bool {
 		if !seen[ref.Name()] {
 			*refs = append(*refs, ref)
 			seen[ref.Name()] = true
 		}
-	}
-	return nil
+		return true
+	})
```

Comment:
The implementation of `func` here seems the same as the one used on line `801`. We could extract this as a standalone func and use the same code in both places.